src/core/server/saved_objects/service/lib/decorate_es_error.ts

```diff
@@ -74,6 +76,12 @@ export function decorateEsError(error: Error) {
   }
 
   if (error instanceof BadRequest) {
+    if (
+      SCRIPT_CONTEXT_DISABLED_REGEX.test(reason || '') ||
+      reason === INLINE_SCRIPTS_DISABLED_MESSAGE
+    ) {
+      return SavedObjectsErrorHelpers.decorateEsCannotExecuteScriptError(error, reason);
+    }
     return SavedObjectsErrorHelpers.decorateBadRequestError(error, reason);
   }
```

Comment:
I added this check for the Bad Request reason so we can fail gracefully if a query's script cannot be executed.

This works but it is brittle, I opened an issue to see if Elasticsearch can add error metadata for us to use instead: elastic/elasticsearch#54012.